segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -386,131 +391,93 @@ public void getStreamSegmentInfo(GetStreamSegmentInfo getStreamSegmentInfo) {
-            int containerId = ((ContainerNotFoundException) u).getContainerId();
-            log.warn("Wrong host. Segment = '{}' (Container {}) is not owned. Operation = '{}').", segment, containerId, operation);
-            connection.send(new WrongHost(requestId, segment, "", clientReplyStackTrace));
-        } else if (u instanceof ReadCancellationException) {
-            log.info("Closing connection {} while reading segment {} due to CancellationException.", connection, segment);
-            connection.send(new SegmentRead(segment, requestId, true, false, EMPTY_BYTE_BUFFER));
-        } else if (u instanceof CancellationException) {
-            log.info("Closing connection {} while performing {} due to {}.", connection, operation, u.getMessage());
-            connection.close();
-        } else if (u instanceof AuthenticationException) {
-            log.warn("Authentication error during '{}'.", operation);
-            connection.send(new WireCommands.AuthTokenCheckFailed(requestId, clientReplyStackTrace));
-            connection.close();
-        } else if (u instanceof UnsupportedOperationException) {
-            log.warn("Unsupported Operation '{}'.", operation, u);
-            connection.send(new OperationUnsupported(requestId, operation, clientReplyStackTrace));
-        } else if (u instanceof BadOffsetException) {
-            BadOffsetException badOffset = (BadOffsetException) u;
-            connection.send(new SegmentIsTruncated(requestId, segment, badOffset.getExpectedOffset(), clientReplyStackTrace));
-        } else {
-            log.error("Error (Segment = '{}', Operation = '{}')", segment, operation, u);
-            connection.close(); // Closing connection should reinitialize things, and hopefully fix the problem
-            throw new IllegalStateException("Unknown exception.", u);
-        }
-
-        return null;
-    }
-
     @Override
-    public void mergeSegments(WireCommands.MergeSegments mergeSegments) {
-        long requestId = mergeSegments.getRequestId();
-        log.debug("Merging Segments {} ", mergeSegments);
+    public void mergeSegments(MergeSegments mergeSegments) {
+        final String operation = "mergeSegments";
 
-        if (!verifyToken(mergeSegments.getSource(), mergeSegments.getRequestId(), mergeSegments.getDelegationToken(), READ_UPDATE,
-                "Merge Segments")) {
+        if (!verifyToken(mergeSegments.getSource(), mergeSegments.getRequestId(), mergeSegments.getDelegationToken(), operation)) {
             return;
         }
 
+        RequestTag requestTag = RequestTracker.initializeAndTrackRequestTag(mergeSegments.getRequestId(), operation,
+                mergeSegments.getSource(), mergeSegments.getTarget());
+        // Merging two segments normally involve sealing and deleting the source segment, so create request tags for this.
+        initializeAllTier2RequestTags("sealSegment", mergeSegments.getSource(), requestTag.getRequestId());
```

Comment:
Why do you do this here? This is not a Tier2-direct operation. Only create and delete are.

Reply:
This has been removed along with all the code related to track Tier-2 activity.